api/src/main/java/io/minio/messages/ListObjectsResult.java

```diff
@@ -87,26 +72,20 @@ public boolean isTruncated() {
     return isTruncated;
   }
 
-  /** Returns List of Items. */
-  public List<Item> contents() {
-    if (contents == null) {
-      return Collections.unmodifiableList(new LinkedList<>());
-    }
-
-    return Collections.unmodifiableList(contents);
+  /** Returns max keys. */
+  public int maxKeys() {
+    return maxKeys;
   }
 
   /** Returns List of Prefix. */
   public List<Prefix> commonPrefixes() {
-    if (commonPrefixes == null) {
-      return Collections.unmodifiableList(new LinkedList<>());
-    }
-
-    return Collections.unmodifiableList(commonPrefixes);
+    return Collections.unmodifiableList(
+        (commonPrefixes == null) ? new LinkedList<>() : commonPrefixes);
   }
 
-  /** Returns continuation token. */
-  public abstract String continuationToken();
+  public List<Item> deleteMarkers() {
+    return deleteMarkers;
+  }
 
-  public abstract String nextContinuationToken();
+  public abstract List<Item> contents();
```

Comment:
```suggestion
  protected <T extends Item> List<T> emptyIfNull(List<T> lst) {
    return Collections.unmodifiableList(MoreObjects.firstNonNull(lst, new LinkedList<T>()));
  }
  public abstract List<Item> contents();
```

I suggest adding a generic method `emptyIfNull()`, which can then be called from the subclass implementations of `contents()`. Note that the return type of the implementations will have to change to the concrete types with this. e.g. `List<Contents>` and `List<Version>`